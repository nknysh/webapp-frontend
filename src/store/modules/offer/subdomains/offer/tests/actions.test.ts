import {
  offerHotelUuidChangeAction,
  offerNameChangeAction,
  offerTermsChangeAction,
  offerFurtherInformationChangeAction,
  offerAddStayBetweenPrerequisiteAction,
  offerRemoveStayBetweenPrerequisiteAction,
  offerChangeStayBetweenPrerequisiteAction,
  offerSetBooleanPrerequisiteAction,
  offerSetPreDiscountAction,
  offerHotelUuidChangeSuccessAction,
} from '../actions';

describe('Offer sub domain actions', () => {
  it('Returns the correct object literals', () => {
    expect(offerHotelUuidChangeAction('UUID123')).toMatchSnapshot();
    expect(offerNameChangeAction('New Name')).toMatchSnapshot();
    expect(offerTermsChangeAction('New Terms')).toMatchSnapshot();
    expect(offerFurtherInformationChangeAction('New Info')).toMatchSnapshot();
    expect(offerAddStayBetweenPrerequisiteAction()).toMatchSnapshot();
    expect(offerRemoveStayBetweenPrerequisiteAction(99)).toMatchSnapshot();
    expect(offerChangeStayBetweenPrerequisiteAction([['2020-01-01', '2020-02-02']])).toMatchSnapshot();
    expect(offerSetBooleanPrerequisiteAction('birthday', null)).toMatchSnapshot();
    expect(offerSetPreDiscountAction(false)).toMatchSnapshot();
    expect(offerHotelUuidChangeSuccessAction('TEST DATA')).toMatchSnapshot();
  });
});
