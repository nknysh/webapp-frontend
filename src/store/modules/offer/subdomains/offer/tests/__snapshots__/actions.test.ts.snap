// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Offer sub domain actions Returns the correct object literals 1`] = `
Object {
  "hotelUuid": "UUID123",
  "type": "offer/OFFER_HOTEL_UUID_CHANGE",
}
`;

exports[`Offer sub domain actions Returns the correct object literals 2`] = `
Object {
  "offerName": "New Name",
  "type": "offer/OFFER_NAME_CHANGE",
}
`;

exports[`Offer sub domain actions Returns the correct object literals 3`] = `
Object {
  "offerTerms": "New Terms",
  "type": "offer/OFFER_TERMS_CHANGE",
}
`;

exports[`Offer sub domain actions Returns the correct object literals 4`] = `
Object {
  "offerFurtherInformation": "New Info",
  "type": "offer/OFFER_FURTHER_INFORMATION_CHANGE",
}
`;

exports[`Offer sub domain actions Returns the correct object literals 5`] = `
Object {
  "type": "offer/OFFER_ADD_STAY_BETWEEN_PREREQUISITE",
}
`;

exports[`Offer sub domain actions Returns the correct object literals 6`] = `
Object {
  "stayBetweenIndex": 99,
  "type": "offer/OFFER_REMOVE_STAY_BETWEEN_PREREQUISITE",
}
`;

exports[`Offer sub domain actions Returns the correct object literals 7`] = `
Object {
  "datesArray": Array [
    Array [
      "2020-01-01",
      "2020-02-02",
    ],
  ],
  "type": "offer/OFFER_CHANGE_STAY_BETWEEN_PREREQUISITE",
}
`;

exports[`Offer sub domain actions Returns the correct object literals 8`] = `
Object {
  "key": "birthday",
  "type": "offer/OFFER_SET_BOOLEAN_PREREQUISITE",
  "value": null,
}
`;

exports[`Offer sub domain actions Returns the correct object literals 9`] = `
Object {
  "type": "offer/OFFER_SET_PRE_DISCOUNT",
  "value": false,
}
`;
